### PR TITLE
chore: add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by Hoverbear**
Also adds the required CNCF COC.

I **much** prefer [Contributor Covenant](https://www.contributor-covenant.org/), 
as it focuses on positivity as well.
But, as a CNCF project we do not have liberty to decide such things.

This is part of fulfilling the [Community](https://github.com/openebs/Mayastor/community)
stuff Github does.

### CODE REVIEW

**Review by blaisedias — APPROVED** [review]
Looks good to me, positive and inviting.

**Review by tiagolobocastro — APPROVED** [review]

**Review by mtzaurus — APPROVED** [review]

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+comments, commits, code, wiki edits, issues, and other contributions that are not
+aligned to this Code of Conduct. By adopting this Code of Conduct, project maintainers
+commit themselves to fairly and consistently applying these principles to every aspect
+of managing this project. Project maintainers who do not follow or enforce the Code of
+Conduct may be permanently removed from the project team.
+
+This code of conduct applies both within project spaces and in public spaces
+when an individual is representing the project or its community.
+
+Instances of abusive, harassing, or otherwise unacceptable behavior in Kubernetes may be reported by contacting the [Kubernetes Code of Conduct Committee](https://git.k8s.io/community/committee-code-of-conduct) via <conduct@kubernetes.io>. For other projects, please contact a CNCF project maintainer or our mediator, Mishi Choudhary <mishi@linux.com>.
```
Comment (gila) [inline CODE_OF_CONDUCT.md:51]: CNCF code of conduct issues should be reported to k8s code of conduct? :thinking:

**Review by gila — CHANGES_REQUESTED** [review]

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (Hoverbear) [reply CODE_OF_CONDUCT.md:51]: Sigh, yes. Yet they use Slack which has no block button. It's so disappointing.

Comment (gila) [reply CODE_OF_CONDUCT.md:51]: hmm OK :) just checking :+1:

**Review by gila — APPROVED** [review]

**Comment by Hoverbear** [comment]
bors merge

**Comment by Hoverbear** [comment]
bors merge

**Pull request merged**